packages/runtime/spec/reflection-merge-partial.spec.ts

```diff
@@ -105,12 +106,12 @@ describe('reflectionMergePartial()', () => {
         describe('and source field empty', () => {
             const source: object = {child: undefined};
             it('does not touch target', () => {
-                const target: any = {child: 123};
+                const target: any = {child: 123, children: []};
```

Comment:
These test are janky. The target/source for these tests do not follow the rules of what should be passed to `reflectionMergePartial()` (e.g. the `MessageInfo.fields` describe `T`). Any `T` should never have optional repeated fields (like `children`). The new implementation of `reflectionMergePartial()` depends on it.